doc/source/conf.py

```diff
@@ -798,3 +799,42 @@ def setup(app):
     app.add_autodocumenter(AccessorMethodDocumenter)
     app.add_autodocumenter(AccessorCallableDocumenter)
     app.add_directive("autosummary", PandasAutosummary)
+
+
+####### This is an ignore list for broken links found in CI run checks for broken-linkcheck.yml
+
+linkcheck_ignore = [
+    re.escape(r'http://'),
+    re.escape(r'http://scatterci.github.io/pydata/pandas'),
+    re.escape(r'http://specs.frictionlessdata.io/json-table-schema/'),
+    re.escape(r'https://'),
```

Comment:
Ignoring the regex engine for now, the matching roughly checks if a given link, e.g. `https://example.com`, matches any of the rules starting from the left. Since you have `'http://'` and `'https://'` in there, every single HTTP link (which should be at least 99% of them) is ignored: https://github.com/pandas-dev/pandas/actions/runs/6506999729/job/17673609837?pr=55246#step:5:147

```suggestion
    re.escape(r'http://scatterci.github.io/pydata/pandas'),
    re.escape(r'http://specs.frictionlessdata.io/json-table-schema/'),
```

Reply:
> Since you have `'http://'` and `'https://'` in there, every single HTTP link (which should be at least 99% of them) is ignored: https://github.com/pandas-dev/pandas/actions/runs/6506999729/job/17673609837?pr=55246#step:5:147

Did not understood this properly. Are you saying that all the `http//` and `https//` links which don't have any host supplied; will be ignored automatically ?

Reply:
Ignoring regex for now, the matching here roughly looks like this:

```python
# This is a link that sphinx finds inside the documentation
import re

def is_ignored(link):
    escaped_link = re.escape(link)

    # linkcheck_ignore is the ignore list in conf.
    for ignore_link_pattern in linkcheck_ignore:
        if escaped_link.startswith(ignore_link_pattern):
             return True
    return False

assert is_ignored("https://example.com")
```

If you have `http://` and `https://` in `linkcheck_ignore` every single link that sphinx finds in the documentation that starts with `http://` and `https://`, which should be almost if not all of them, will be ignored. And that is what you see in the CI output. 

But this is not what we want. We only want to ignore the broken ones. Otherwise the whole linkcheck does effectively nothing.